src/sql/src/plan.rs

```diff
@@ -76,11 +76,9 @@ pub enum Plan {
     CreateRole {
         name: String,
     },
-    CreateSource {
-        name: FullName,
-        source: Source,
-        if_not_exists: bool,
-        materialized: bool,
+    CreateSource(CreateSourcePlan),
```

Comment:
Why not only have the plural?

Reply:
Reading further this seems like there is some hidden invariant driving this design?